### PR TITLE
[ISSUE #4029] api and api_rule_relation mapper

### PR DESCRIPTION
**Pull request by 847850277**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: db/init/mysql/schema.sql [hunk 1]
```diff
@@ -39,6 +39,52 @@ CREATE TABLE `alert_template`  (
...
+                       `version` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'api version,for example V0.01',
+                       `rpc_type` varchar(64) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'http,dubbo,sofa,tars,websocket,springCloud,motan,grpc',
+                       `status` tinyint(4) NOT NULL COMMENT '0-unpublished1-published2-offline',
+                       `ext` varchar(1024) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'extended fields',
+                       `owner` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'owner',
+                       `api_desc` varchar(1024) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'the api description',
+                       `document` text COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'complete documentation of the api, including request parameters and response parameters',
+                       `document_md5` char(32) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'document_md5',
+                       `source` int(0) NOT NULL COMMENT '0-swagger,1-annotation generation,2-create manuallym,3-import swagger,4-import yapi',
```
Comment (dragon-zhang) [inline db/init/mysql/schema.sql:61]: `source` is a keyword, change it to `api_source`.

Comment (847850277) [reply db/init/mysql/schema.sql]: fix it

File: db/init/mysql/schema.sql [hunk 2]
```diff
@@ -39,6 +39,52 @@ CREATE TABLE `alert_template`  (
...
+CREATE TABLE `api` (
+                       `id` varchar(128) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'primary key id',
+                       `context_path` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'the context_path',
+                       `path` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'the path',
+                       `http_method` int(0) NOT NULL COMMENT '0-get,1-head,2-post,3-put,4-patch,5-delete,6-options,7-trace',
+                       `consume` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'consume',
+                       `produce` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'produce',
+                       `version` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'api version,for example V0.01',
+                       `rpc_type` varchar(64) COLLATE utf8mb4_unicode_ci NOT NULL COMMENT 'http,dubbo,sofa,tars,websocket,springCloud,motan,grpc',
+                       `status` tinyint(4) NOT NULL COMMENT '0-unpublished1-published2-offline',
```
Comment (dragon-zhang) [inline db/init/mysql/schema.sql:55]: please change `status` to `state`.

**Comment by li-keguo** [comment]
format ddl

**Comment by 847850277** [comment]
> format ddl

fixed it

File: db/init/mysql/schema.sql (hunk 2 above)
Comment (847850277) [reply db/init/mysql/schema.sql]: fixed it

File: db/init/pg/create-table.sql [hunk 3]
```diff
@@ -47,6 +47,70 @@ MAXVALUE 9223372036854775807
...
+-- ----------------------------
+-- Table structure for api_rule_relation
+-- ----------------------------
+DROP TABLE IF EXISTS "public"."api_rule_relation";
+CREATE TABLE "public"."api_rule_relation" (
+                                "id"           varchar(128) COLLATE "pg_catalog"."default" NOT NULL,
+                                "api_id"       varchar(128) COLLATE "pg_catalog"."default" NOT NULL,
+                                "rule_id"      varchar(128) COLLATE "pg_catalog"."default" NOT NULL,
+                                "date_created" timestamp(6) NOT NULL DEFAULT timezone('UTC-8'::text, (now())::timestamp(0) without time zone),
+                                "date_updated" timestamp(6) NOT NULL DEFAULT timezone('UTC-8'::text, (now())::timestamp(0) without time zone)
```
Comment (dragon-zhang) [inline db/init/pg/create-table.sql:105]: needs format here.

**Review by dragon-zhang — APPROVED** [review]

**Pull request merged**